base/audit_events.go

```diff
@@ -72,6 +240,638 @@ var AuditEvents = events{
+		Description: "Information about this user was viewed",
+		MandatoryFields: AuditFields{
+			"username": "username",
+			"db":       "database name",
+		},
+		EnabledByDefault:   true,
+		FilteringPermitted: false,
+		EventType:          eventTypeAdmin,
+	},
+	AuditIDUserUpdate: {
+		Name:        "Update user",
+		Description: "User was updated",
+		MandatoryFields: AuditFields{
+			"username": "username",
+			"db":       "database name",
+			"roles":    []string{"list", "of", "roles"},
+			"channels": []string{"list", "of", "channels"},
+		},
+		EnabledByDefault:   true,
+		FilteringPermitted: false,
+		EventType:          eventTypeAdmin,
+	},
+	AuditIDUserDelete: {
+		Name:        "Delete user",
+		Description: "User was deleted",
+		MandatoryFields: AuditFields{
+			"username": "username",
+			"db":       "database name",
+		},
+		EnabledByDefault:   true,
+		FilteringPermitted: false,
+		EventType:          eventTypeAdmin,
+	},
+	AuditIDRoleCreate: {
+		Name:        "Create role",
+		Description: "A new role was created",
+		MandatoryFields: AuditFields{
+			"role":           "role_name",
+			"db":             "database name",
+			"admin_channels": []string{"list", "of", "channels"},
+		},
+		EnabledByDefault:   true,
+		FilteringPermitted: false,
+		EventType:          eventTypeAdmin,
+	},
+	AuditIDRoleRead: {
+		Name:        "Read role",
+		Description: "Information about this role was viewed",
+
+		MandatoryFields: AuditFields{
+			"role": "role_name",
+			"db":   "database name",
+		},
+		EnabledByDefault:   true,
+		FilteringPermitted: false,
+		EventType:          eventTypeAdmin,
+	},
+	AuditIDRoleUpdate: {
+		Name:        "Update role",
+		Description: "Role was updated",
+
+		MandatoryFields: AuditFields{
+			"role":           "role_name",
+			"db":             "database name",
+			"admin_channels": []string{"list", "of", "channels"},
+		},
+		EnabledByDefault:   true,
+		FilteringPermitted: false,
+		EventType:          eventTypeAdmin,
+	},
+	AuditIDRoleDelete: {
+		Name:        "Delete role",
+		Description: "Role was deleted",
+		MandatoryFields: AuditFields{
+			"role": "role_name",
+			"db":   "database name",
+		},
+		EnabledByDefault:   true,
+		FilteringPermitted: false,
+		EventType:          eventTypeAdmin,
+	},
+	AuditIDChangesFeedStarted: {
+		Name:        "Changes feed started",
+		Description: "Changes feed was started",
+		MandatoryFields: AuditFields{
+			"db":    "database name",
+			"ks":    "keyspace",
+			"since": "since",
+		},
+		OptionalFields: AuditFields{
+			"filter":    "filter",
+			"doc_ids":   []string{"list", "of", "doc_ids"},
+			"channels":  []string{"list", "of", "channels"},
+			"feed_type": "continuous, normal, longpoll, websocket, etc.",
+		},
+		EnabledByDefault:   true,
+		FilteringPermitted: true,
+		EventType:          eventTypeData,
+	},
+	AuditIDReplicationConnect: {
+		Name:        "Replication connect",
+		Description: "A client connected using WebSocket/BLIP for replication",
```

Comment:
I was tempted to use a friendly name of Couchbase Lite/Inter-Sync Gateway here.

Reply:
made this more general and can be more specific with the optional client_type and client_version fields.